src/schedule/zephyr_dp_schedule.c

```diff
@@ -240,11 +251,13 @@ void scheduler_dp_ll_tick(void *receiver_data, enum notify_id event_type, void *
 							       mod->num_of_sinks);
 			if (mod_ready) {
 				/* set a deadline for given num of ticks, starting now */
-				k_thread_deadline_set(pdata->thread_id, pdata->period_clock_ticks);
+				k_thread_deadline_set(pdata->thread_id,
+						      pdata->deadline_clock_ticks);
 
 				/* trigger the task */
 				curr_task->state = SOF_TASK_STATE_RUNNING;
 				k_sem_give(&pdata->sem);
+				pdata->ll_cycles_to_deadline = pdata->deadline_ll_cycles;
```

Comment:
I think this should go before the semaphore to avoid a race

Reply:
does not matter, the semaphore give is releasing a module thread, which has no access to pdata-> context.  The context itself is protected by scheduler_dp_lock / scheduler_dp_unlock

Anyway I see CI is still stalled so I have to retrigger anyway - so I'll change it because it looks suspicious at first